R/estimateDiversity.R

```diff
@@ -254,7 +254,11 @@ setMethod("estimateDiversity", signature = c(x="SummarizedExperiment"),
         }
 
         # input check
-        index<- match.arg(index, several.ok = TRUE)
+        supported_index <- c("coverage", "fisher", "gini_simpson", 
+                             "inverse_simpson", "log_modulo_skewness", "shannon")
+        if( !all( index %in% supported_index ) ){
+            stop("'index' contains unsupported indices.", call. = FALSE)
+        }
```

Comment:
Looks good! Also check that the length of index is over 0 (it is not checked currently. NULL might cause problems --> test `index = NULL`